### PR TITLE
Release bugfix

### PR DESCRIPTION
**Pull request by javydreamercsw**
This is still a WIP so don't merge yet. Having issues testing due to the lack of a Windows machine.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,22 +2,55 @@
...
-  - oraclejdk8
-script: ./gradlew clean build test compileSlowtest datatest pfinttest allReports
+
+compiler:
+    - gcc
+os:
+    - linux
+    jdk: oraclejdk8
+    - osx
+    osx_image: xcode8
```
Comment (javydreamercsw) [inline .travis.yml]: Build both in linux and Mac.

File: .travis.yml [hunk 2]
```diff
@@ -2,22 +2,55 @@
...
- - sudo apt-get install lib32z1 lib32ncurses5 nsis perl
- - export PERL_MM_USE_DEFAULT=1
- - sudo cpan Readonly
- - chmod +x gradlew
+    - sudo apt-get update
+    - sudo apt-get install lib32z1 lib32ncurses5 nsis perl
+    - export PERL_MM_USE_DEFAULT=1
+    - sudo cpan Readonly
+    - chmod +x gradlew
+    - if [[ $TRAVIS_OS_NAME == 'linux' ]]; then sudo apt-get install scons mingw32 mingw32-binutils zlib1g-dev libcppunit-dev; fi
```
Comment (javydreamercsw) [inline .travis.yml]: Install NSIS.

File: .travis.yml [hunk 3]
```diff
@@ -2,22 +2,55 @@
...
+    - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+    - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 cache:
   directories:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
+    
+before_deploy:
+    # Build release files
+    - ./gradlew pcgenRelease
```
Comment (javydreamercsw) [inline .travis.yml]: Create release files on deploy.

File: .travis.yml [hunk 4]
```diff
@@ -2,22 +2,55 @@
...
+
+deploy:
+  provider: releases
+  #Replace this with a real token
+  api_key: "GITHUB OAUTH TOKEN"
```
Comment (javydreamercsw) [inline .travis.yml]: This needs to be obtained and replaced in the file.

**Comment by javydreamercsw** [comment]
Pending further tests before merge everything seems to work fine for me both on Mac and Windows hosts.

Please run thru the release wiki instructions before merging.

File: .travis.yml [hunk 5]
```diff
@@ -2,22 +2,55 @@
...
 dist: trusty
-jdk:
-  - oraclejdk8
-script: ./gradlew clean build test compileSlowtest datatest pfinttest allReports
+
+matrix:
+  include:
+    - os: linux
+      jdk: oraclejdk8
+  allow_failures:
```
Comment (javydreamercsw) [inline .travis.yml]: Ignore Mac failures.

**Comment by LegacyKing** [comment]
Looks good to me. Tested and no issues in Netbeans. Though I'm missing the two custom folders. Which I can add back in.

**Comment by LegacyKing** [comment]
Disregard the two custom folder comment. Looks good! Now to grab Linux and see if the pcgen.sh is working.

**Comment by javydreamercsw** [comment]
Great to know. Let me know how it does. BTW, I didn't make any change to fix the Linux script. It was already there in one of the tasks. I assume it was copied by another task that didn't have that change.

**Pull request merged**